pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go

```diff
@@ -156,12 +160,16 @@ func applyLocalityFailover(
 	locality *core.Locality,
 	loadAssignment *endpoint.ClusterLoadAssignment,
 	failover []*v1alpha3.LocalityLoadBalancerSetting_Failover,
+	prioritiesAssigned int,
 ) {
 	// key is priority, value is the index of the LocalityLbEndpoints in ClusterLoadAssignment
 	priorityMap := map[int][]int{}
 
 	// 1. calculate the LocalityLbEndpoints.Priority compared with proxy locality
 	for i, localityEndpoint := range loadAssignment.Endpoints {
+		if localityEndpoint.Priority < uint32(prioritiesAssigned-1) {
```

Comment:
Can you elaborate what does this mean

Reply:
So since the `failover` is used for only the endpoints with least priority all the localityEndpoints which have already been assigned priority will be not taken into account. This is to prevent `Failover` setting to step on the foot of `failoverPriority`.